#!/usr/bin/env python
PACKAGE = "mrs_controllers"

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kpxy", double_t, 1, "Position constant for xy-axes", 0.0, 0.0, 1.0)
gen.add("kvxy", double_t, 2, "Velocity constant for xy-axes", 0.0, 0.0, 1.0)
gen.add("kixy", double_t, 4, "Integral constant for xy-axes", 0.0, 0.0, 1.0)
gen.add("kpz", double_t, 8, "Position constant for z-axis", 0.0, 0.0, 1.0)
gen.add("kvz", double_t, 16, "Velocity constant for z-axis", 0.0, 0.0, 1.0)
gen.add("kixy_lim", double_t, 32, "xy-axes integral limit", 0.0, 0.0, 10.0)
gen.add("km", double_t, 64, "Integral constant for mass", 0.0, 0.0, 2.0)
gen.add("km_lim", double_t, 128, "mass integral limit", 0.0, 0.0, 2.0)
gen.add("yaw_offset", double_t, 256, "yaw offset", 0.0, -30.0, 30.0)

exit(gen.generate(PACKAGE, "LsfController", "lsf_gains"))
